File: ecom.api/src/main/java/aditi/wing/ecom/api/domain/seller/service/SellerService.java
package aditi.wing.ecom.api.domain.seller.service;

import java.util.List;
import java.util.UUID;

import aditi.wing.ecom.api.domain.seller.dto.BulkStockUpdateDto;
import aditi.wing.ecom.api.domain.seller.dto.DashboardStatsDto;
import aditi.wing.ecom.api.domain.seller.dto.PayoutDto;
import aditi.wing.ecom.api.domain.seller.dto.ProductRequestDto;
import aditi.wing.ecom.api.domain.seller.dto.ProductResponseDto;

public interface SellerService {

    // Dashboard
    DashboardStatsDto getDashboardStats(UUID sellerId);

    // Products
    ProductResponseDto createProduct(UUID sellerId, ProductRequestDto request);

    List<ProductResponseDto> getSellerProducts(UUID sellerId);

    ProductResponseDto updateProduct(UUID sellerId, UUID productId, ProductRequestDto request);

    void deleteProduct(UUID sellerId, UUID productId);

    // Inventory
    List<ProductResponseDto> getLowStockProducts(UUID sellerId);

    void bulkUpdateStock(UUID sellerId, BulkStockUpdateDto request);

    // Payouts
    List<PayoutDto> getPayouts(UUID sellerId);
}

File: web/lib/services/seller.service.ts
import { apiClient, API_CONFIG } from "@/lib/api-client";
import type {
  Product,
  ProductRequest,
  DashboardStats,
  Payout,
  BulkStockUpdate,
  Category,
} from "@/lib/types/seller";

export class SellerService {
  /**
   * Get dashboard statistics
   */
  async getDashboardStats(): Promise<DashboardStats> {
    return apiClient.get<DashboardStats>(API_CONFIG.endpoints.seller.dashboard);
  }

  /**
   * Create new product
   */
  async createProduct(data: ProductRequest): Promise<Product> {
    return apiClient.post<Product>(API_CONFIG.endpoints.seller.products, data);
  }

  /**
   * Upload product image
   */
  async uploadProductImage(file: File): Promise<{ imageUrl: string }> {
    const formData = new FormData();
    formData.append("file", file);
    return apiClient.uploadFile<{ imageUrl: string }>(
      API_CONFIG.endpoints.seller.uploadImage,
      formData,
    );
  }

  /**
   * Get all seller products
   */
  async getProducts(): Promise<Product[]> {
    return apiClient.get<Product[]>(API_CONFIG.endpoints.seller.products);
  }

  /**
   * Update product
   */
  async updateProduct(
    productId: string,
    data: ProductRequest,
  ): Promise<Product> {
    return apiClient.put<Product>(
      `${API_CONFIG.endpoints.seller.products}/${productId}`,
      data,
    );
  }

  /**
   * Delete product
   */
  async deleteProduct(productId: string): Promise<void> {
    return apiClient.delete(
      `${API_CONFIG.endpoints.seller.products}/${productId}`,
    );
  }

  /**
   * Get low stock products
   */
  async getLowStockProducts(): Promise<Product[]> {
    return apiClient.get<Product[]>(API_CONFIG.endpoints.seller.lowStock);
  }

  /**
   * Bulk update stock quantities
   */
  async bulkUpdateStock(data: BulkStockUpdate): Promise<void> {
    return apiClient.post(API_CONFIG.endpoints.seller.bulkUpdate, data);
  }

  /**
   * Get payouts
   */
  async getPayouts(): Promise<Payout[]> {
    return apiClient.get<Payout[]>(API_CONFIG.endpoints.seller.payouts);
  }

  // Get Category

  async getCategories(): Promise<Category[]> {
    return apiClient.get<Category[]>(API_CONFIG.endpoints.seller.categories);
  }
}

export const sellerService = new SellerService();
